src/web/demo/src/main/java/org/geoserver/web/demo/DemoRequestResponse.java

```diff
@@ -51,7 +52,7 @@ public DemoRequestResponse(final IModel model) {
         form.add(new HiddenField("url", new PropertyModel(model, "requestUrl")));
         form.add(new TextArea("body", new PropertyModel(model, "requestBody")));
         form.add(new HiddenField("username", new PropertyModel(model, "userName")));
-        form.add(new HiddenField("password", new PropertyModel(model, "password")));
+        form.add(new HiddenField("password", new Model(((DemoRequest)model.getObject()).getPassword())));
```

Comment:
Confused as to why username interpolation works, but password one needs to dodge the model instead?

Reply:
I honestly don't know, and didn't dig into Wicket enough to find out.

Reply:
>Note that by default the model object is nullified after each request to prevent the entered password to be serialized along with the containing page, see{@link#setResetPassword(boolean)}. Hmm, maybe something to do with that? I'll check.

Which... now that I think about it makes perfect sense, and also means that this page is broken as is and really should be fixed properly. This page sends the submitted password back to fill the modal window that pops up which we really shouldn't be doing. Wicket was trying to be helpful the whole time I guess.